pkg/vkMachinery/csr/approve.go

```diff
@@ -39,5 +70,48 @@ func WatchCSR(clientset k8s.Interface, label string, resyncPeriod time.Duration)
 		},
 	})
 
-	go informer.Start(stop)
+	go informer.Start(stop.Done())
 }
+
+// WaitForApproval returns a CRT when available for a specific CSR resource. It timeouts after a while, if the CRT is not available.
+func ForgeInformer(cancel context.CancelFunc, client k8s.Interface, name string, crtChan chan []byte) cache.SharedIndexInformer {
+	var resyncPeriod = 10 * time.Second
+	var fieldSelector = map[string]string{
+		"metadata.name": name,
+	}
+	lo := func(options *metav1.ListOptions) {
+		options.FieldSelector = labels.SelectorFromSet(fieldSelector).String()
+	}
+	options := []informers.SharedInformerOption{
+		informers.WithTweakListOptions(lo),
+	}
+	informer := informers.NewSharedInformerFactoryWithOptions(client, resyncPeriod, options...)
```

Comment:
```suggestion
	informer := informers.NewSharedInformerFactoryWithOptions(client, resyncPeriod, informers.WithTweakListOptions( func(options *metav1.ListOptions) {
		options.FieldSelector = labels.SelectorFromSet(fieldSelector).String()
	}))
```

Should be equivalent and less verbose